abci/client/client.go

```diff
@@ -33,35 +33,24 @@ type Client interface {
 
 	// Asynchronous requests
 	FlushAsync(context.Context) (*ReqRes, error)
-	EchoAsync(ctx context.Context, msg string) (*ReqRes, error)
-	InfoAsync(context.Context, types.RequestInfo) (*ReqRes, error)
 	DeliverTxAsync(context.Context, types.RequestDeliverTx) (*ReqRes, error)
 	CheckTxAsync(context.Context, types.RequestCheckTx) (*ReqRes, error)
```

Comment:
I believe this gets used in the mempool right? Is it possible (in a follow up PR of course) to modify the usage so we can call `CheckTx` instead?

Reply:
Yeah, I tried this, sort of naively, and I think it's possible, but the impact on the mempool is pretty good (mostly because there are a bunch of tests that do assertions/etc. via the callback. I think the best path forward would be to merge this, and then write a shim in the mempool package that does the switch and try and go from there.